File: JAVA-Basic/JavaBasic/src/WhileLoop/Exercise/ExamPreparation.java
package WhileLoop.Exercise;

import java.util.Scanner;

public class ExamPreparation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int badGrades = Integer.parseInt(scan.nextLine());
        double gradeSum = 0;
        int badGradesCounter = 0;
        int gradeCounter = 0;
        String lastExerciseName = "";

        String exerciseName = scan.nextLine();

        while (!exerciseName.equals("Enough")) {
            int currentGrade = Integer.parseInt(scan.nextLine());
            gradeSum += currentGrade;
            gradeCounter++;

            if (currentGrade <= 4) {
                badGradesCounter++;
            }
            if (badGradesCounter >= badGrades) {
                System.out.printf("You need a break, %d poor grades.", badGrades);
                return;
            }
            lastExerciseName = exerciseName;
            exerciseName = scan.nextLine();
        }
        System.out.printf("Average score: %.2f\n", gradeSum / gradeCounter);
        System.out.printf("Number of problems: %d\n", gradeCounter);
        System.out.printf("Last problem: %s", lastExerciseName);
    }
}

File: 01.JS-Basic/WHILE-ЦИКЪЛ Упражнение/ExamPreparation.js
function examPreparation(input) {

    let maximumBad = Number(input[0]);
    let badScoreCount = 0
    let allScoreCount = 0
    let lastExercies = ""
    let index = 1
    let scoreSum = 0

    while (maximumBad != badScoreCount) {
        let current = input[index]

        if (current === "Enough") {
            console.log(`Average score: ${(scoreSum / allScoreCount).toFixed(2)}`)
            console.log(`Number of problems: ${allScoreCount}`)
            console.log(`Last problem: ${lastExercies}`)
            break;
        }
        else if (index % 2 != 0) {
            lastExercies = current
            allScoreCount++
        }
        else {
            if (current <= 4) {
                badScoreCount++
                scoreSum += Number(current)
            } else {
                scoreSum += Number(current)
            }
        }
        if (maximumBad === badScoreCount) {
            console.log(`You need a break, ${badScoreCount} poor grades.`)
        }
        index++
    }

}
examPreparation(["2",
"Income",
"3",
"Game Info",
"6",
"Best Player",
"4"])

